cmd/thanos/receive.go

```diff
@@ -1097,6 +1136,8 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 		Default("10000").Uint64Var(&rc.topMetricsMinimumCardinality)
 	cmd.Flag("receive.top-metrics-update-interval", "The interval at which the top metrics are updated.").
 		Default("5m").DurationVar(&rc.topMetricsUpdateInterval)
+	cmd.Flag("receive.store-matcher-converter-cache-capacity", "The number of label matchers to cache in the matcher converter for the Store API. Set to 0 to disable to cache. Default is 0.").
```

Comment:
Curious how do we decide this cache size about how many matcher in each region?

Reply:
We use a constant 30k cache size across all regions. I tried 2k, 30k and 100k, and 30k gives great cache hit rate (99.8%) with negligible memory footprint in oregon-dev.